### DISCUSSION TITLE
Hot Reload

### DISCUSSION THREAD

**Discussion by Ayush-vachhani** [AvaloniaUI/Avalonia#14098] [Q&A]
Does avalonia actually support hot reloading, i doesn't seem to work for me, also is hot reloading supported for both UI and the logic?

**Comment by timunie** [comment]
planned but not working out of the box, see: #3266

**Comment by thevortexcloud** [comment]
It's apparently going to be a paid feature: https://github.com/AvaloniaUI/Avalonia/discussions/14097

**Comment by jpgarza93** [comment]
HotReload would be a GAME CHANGER. So much extra time spent due to this not being available. I am sure there is a reason why VS does not charge for it.

**Comment by MikeCodesDotNET** [comment]
We do not plan to implement HotReload in the foreseeable future as it would require extensive engineering effort.  

As @grokys said in the [previously linked discussion](https://github.com/AvaloniaUI/Avalonia/discussions/14097#discussioncomment-8031373), if we were to implement HotReload, it would have to be a paid feature.

**Reply by jpgarza93** [reply]
Why not work together with [HotAvalonia ](https://github.com/Kir-Antipov/HotAvalonia) or [Live.Avalonia](https://github.com/AvaloniaUI/Live.Avalonia)? Combined efforts might simplify things?

My issue with both of those is that I can't get them to work with my avalonia UI cross-platform mvvm app

But even if it's a paid feature I'm sure it will be a no-brainer for people coming from WPF.

**Reply by waliqadri101** [reply]
Were you able to make it work? It worked for me, but only for the XAML part. Which is still mind blowing, Any changes in the code still had to be recompiled.

**Reply by stevemonaco** [reply]
HotAvalonia works well for me on desktop (VS or Rider) as long as you avoid private event handlers. Any code file modifications would need to need to go through .NET Hot Reload as HotAvalonia doesn't process C# files.

**Reply by jpgarza93** [reply]
Yeah, HotAvalonia works great. Between that for xaml and regular .NET hot reload for C# you can get a better flow when developing your app

**Comment by grokys** [comment]
For anyone following this discussion, we've released hot reload support today: https://avaloniaui.net/blog/hot-reload-ga
